streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -372,15 +371,13 @@ public static StreamThread create(final TopologyMetadata topologyMetadata,
                                       final Runnable shutdownErrorHook,
                                       final BiConsumer<Throwable, Boolean> streamsUncaughtExceptionHandler) {
 
-        final boolean stateUpdaterEnabled = InternalConfig.stateUpdaterEnabled(config.originals());
-
         final String threadId = clientId + THREAD_ID_SUBSTRING + threadIdx;
         final String stateUpdaterId = threadId.replace(THREAD_ID_SUBSTRING, STATE_UPDATER_ID_SUBSTRING);
-        final String restorationThreadId = stateUpdaterEnabled ? stateUpdaterId : threadId;
+        final String restorationThreadId = stateUpdaterId;
```

Comment:
You could directly use `stateUpdaterId` instead of `restorationThreadId` since the distinction between state updater ID and thread ID for restoration does not hold anymore.